### PR TITLE
fix: check if newly installed extensions dependencies are already installed

### LINKED ISSUE #9069: Cannot install an extension with dependency on another
### Bug description

I tried to install OpenShift Local extension and I got the error message:

`Error: Missing dependencies for this extension: redhat.redhat-authentication`

But the extension is installed

### Operating system

Win11Pro

### Installation Method

None

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SoniaSandler**
### What does this PR do?
This PR fixes the dependency check for extensions that were installed after starting PD.

### What issues does this PR fix or reference?
Closes https://github.com/containers/podman-desktop/issues/9069

### How to test this PR?

- [X] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/extension-loader.ts [hunk 1]
```diff
@@ -493,21 +493,24 @@ export class ExtensionLoader {
...
+        if (!existingExtensions.find(existingExtension => existingExtension.id === dependency)) {
           missingDependencies.push(dependency);
         }
       });
       extension.missingDependencies = missingDependencies;
     });
   }
 
-  async loadExtensions(analyzedExtensions: AnalyzedExtension[]): Promise<void> {
+  async loadExtensions(analyzedExtensions: AnalyzedExtension[], onStart: boolean): Promise<void> {
```
Comment (benoitf) [inline packages/main/src/plugin/extension-loader.ts]: issue(blocking): no parameter needs to be added there

all consumers ask to load extensions, no 'onStart' (or another name paramerer)

File: packages/main/src/plugin/extension-loader.ts [hunk 2]
```diff
@@ -494,10 +494,11 @@ export class ExtensionLoader {
   // check if all dependencies are available
   // if not, set the missingDependencies property
   searchForMissingDependencies(analyzedExtensions: AnalyzedExtension[]): void {
+    const existingExtensions = [...new Set([...this.analyzedExtensions.values(), ...analyzedExtensions])];
```
Comment (benoitf) [inline packages/main/src/plugin/extension-loader.ts]:
```suggestion
    const existingExtensions = Array.from(new Set([...this.analyzedExtensions.values(), ...analyzedExtensions]));
```

Comment (benoitf) [reply packages/main/src/plugin/extension-loader.ts]: suggestion: use the Array.from syntax

**Review by benoitf — APPROVED** [review]
thanks for the changes

File: packages/main/src/plugin/extension-loader.ts (hunk 2 above)
Comment (SoniaSandler) [reply packages/main/src/plugin/extension-loader.ts]: done

**Review by axel7083 — APPROVED** [review]

**Pull request merged**